### PR TITLE
Restructure the dashboard a bit and update it to use FontAwesome instead of Glyphicon

### PR DESCRIPTION
**Pull request by blag**
I tweaked the look of the dashboard. The black boxes are just to keep my internal names out of the public record.

Additionally, I filled out `watchman/base.html` with basic HTML that should more closely mimic an actual site's `base.html`, and had `watchman/dashboard.html` extend from that. That way it should be easier to embed the dashboard in an existing site that uses Bootstrap components (by just overwriting `watchman/base.html`).

~~The only logic change that could affect anything else is I changed the check names to be singular instead of plural. I haven't tested that part yet.~~
Travis tested it for me. It broke stuff. So now I just strip off the 's' at the end of plural type names and stuff it into `type_singular` for the template.

Status: OK
----------

The overall system status is now on the upper right with a FontAwesome icon (instead of Glyphicon because Bootstrap is moving to FontAwesome).

watchman-status-ok

Status: Error
-------------

I separated the error text and the button for tracebacks so I could get rid of the explanatory text at the bottom. I figure if an interface/widget needs explanatory text it's probably not that great to begin with. This way everything is more direct and explicit (click "Traceback" to get a traceback, status text is just text, not a button).

Also, if not all subsystems of a specific type are all up, they **all** get a yellow background except for the subsystem that is down. That is more confusing in words, the picture should explain enough.

watchman-status-error

Traceback
---------

I tweaked the traceback so its width auto-expands to fill the whole window for all screen sizes (margins are 10px - Bootstrap's default - when the screen is <= 768px wide and 30px otherwise).

watchman-traceback

### CODE REVIEW

**Comment by mwarkentin** [comment]
@blag Thanks for another contribution! I'll try to review soon!

**Comment by blag** [comment]
Really, the only line that has any logic changes is this one where I'm constructing the template context in `watchman/views.py`:

```python
...
def dashboard(request):
    ...
                check_types.append({
                    'type': _type,
                    'type_singular': _type[:-1] if _type.endswith('s') else _type,  # <-- this one
                    'ok': type_overall_status,
                    'statuses': statuses})
  ```

Other than that it's all changes to the HTML. And I'm already overriding `watchman/base.html` for my site to add a navigation bar to the top, which is exactly the functionality I wanted to enable. But the dashboard should work just fine for people.

So this should be pretty easy to review. 😄

File: watchman/templates/watchman/base.html [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
-            <script src="{% static 'js/ninja.js' %}" type="text/javascript"></script>
-        {% endblock javascript %}
-
-    {% endblock extra_js %}
-{% endcomment %}
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
```
Comment (mwarkentin) [inline watchman/templates/watchman/base.html]: Worth bumping to [`3.3.7`](https://github.com/twbs/bootstrap/releases/tag/v3.3.7) while were poking around in here?

**Comment by mwarkentin** [comment]
Overall, this looks good. If you can bump up the bootstrap version, then I'll pull it in.

File: watchman/templates/watchman/base.html (hunk 1 above)
Comment (blag) [inline watchman/templates/watchman/base.html]: 👍  Good thinking! Done.

**Comment by blag** [comment]
Bumped Bootstrap to 3.3.7. 👍

**Comment by mwarkentin** [comment]
LGTM - hoping to get a `0.11` release out shortly which will include this.

**Comment by mwarkentin** [comment]
@blag I've cut a 0.11rc here: https://pypi.python.org/pypi/django-watchman

Please give it a shot and let me know if you discover any issues. I'll cut the official release in a day or two assuming that I don't hear anything.

**Comment by mwarkentin** [comment]
@blag I've released 0.11 including these changes: https://pypi.python.org/pypi/django-watchman

**Pull request merged**